src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -240,6 +236,11 @@ def tdr(self):
     def _tdr(cls):
         return TDRClient.for_indexer()
 
+    def verify_source(self, ref: TDRSourceRef) -> None:
+        actual_name = self.tdr.lookup_source_name(ref)
+        require(ref.spec.name == actual_name,
+                'Source name changed unexpectedly', ref, actual_name)
```

Comment:
I think we should move the implementation to the client.

```suggestion
        self.tdr.verify_source((ref)
```

and inline `lookup_source_name`.